src/Reflection/ReflectionNamedType.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\BetterReflection\Reflection;
+
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+
+use function array_key_exists;
+use function strtolower;
+
+class ReflectionNamedType extends ReflectionType
+{
+    private const BUILT_IN_TYPES = [
+        'int'      => null,
+        'float'    => null,
+        'string'   => null,
+        'bool'     => null,
+        'callable' => null,
+        'self'     => null,
+        'parent'   => null,
+        'array'    => null,
+        'iterable' => null,
+        'object'   => null,
+        'void'     => null,
+        'mixed'    => null,
+        'static'   => null,
+        'null'     => null,
+    ];
+
+    private string $name;
+
+    /**
+     * @param Identifier|Name $type
```

Comment:
This type can (and should) be declared natively

Reply:
It will be automatically converted by https://github.com/Roave/BetterReflection/pull/753